test/core/networking/server_test.dart

```diff
@@ -26,7 +26,35 @@ void main() {
 
     // Wait for handler to be called and check
     await received.awaitValue(Duration(seconds: 5), raiseOnTimeout: true);
-    received.expect(equals(true));
+    received.expect(isTrue);
+
+    // Teardown
+    server.close();
+  });
+
+  test('WebServer can receive https request', () async {
+    final VHook<bool?> received = VHook<bool?>(null);
+
+    final securityContext = SecurityContext()
+      ..useCertificateChain('./test/helpers/mocks/certs/mock_key_store.p12')
+      ..usePrivateKey('./test/helpers/mocks/certs/mock_key.pem');
+
+    final WebServer server = await openWebServer(securityContext: securityContext);
+    server.httpOnRequest = (HttpRequest req) {
+      received.set(true);
+      req.response.statusCode = 404;
+      req.response.close();
+    };
+
+    // Create HttpClient that accepts self-signed certificate and connect to server
+    final client = HttpClient()
+      // Trust every certificate
+      ..badCertificateCallback = ((X509Certificate cert, String host, int port) => true);
```

Comment:
Why do you add the type name when you are not even going to use them?

Reply:
Because I hate dynamic

Reply:
Yes you are. The parameters are already typed by the definition of the callback function. You are only repeating the type, but do not change or set it in any way. When actually using the callback I can understand that you might want to do it for readability. But in cases like this, I do not recommend it. 
A known example might be the callback of the `map` function. If you write 
```dart
['hello', 'world'].map((e) => '$e!')
```
e is always String and you don't have to write 
```dart
['hello', 'world'].map((String e) => '$e!'`)
```

e is implicitly typed

Reply:
Yes, but if we change the interface sometime, this would not instantly raise a compiler error. Explicit types are visually appealing and helpful.

Reply:
I don't follow you on the first part, bc implicit types will have the same effect, but I also don't mind if you continue this way.